src/draw.c

```diff
@@ -501,7 +512,7 @@ void draw_rounded_rect(cairo_t *c, int x, int y, int width, int height, int corn
 
         cairo_new_sub_path(c);
 
-        if (last) {
+        if (last || settings.gaps) {
```

Comment:
I think it's better to pass `first = true` and `last = true` to this function when gaps is enabled. This is true in the way that it's the last in a block of notifications.
The same counts for the other functions that get passed a first and last variable.